dwave/system/composites/embedding.py

```diff
@@ -515,6 +516,186 @@ def sample(self, bqm, chain_strength=1.0, chain_break_method=None,
             chain_break_fraction=chain_break_fraction, **parameters)
 
 
+class ParamEmbeddingComposite(dimod.ComposedSampler):
+    """Composite that maps problems to a structured sampler using a
+    parameterized embedding method.
+
+    Args:
+       sampler (dimod.Sampler)
+            Structured dimod sampler.
+
+       embedding_method (object, optional, default=minorminer)
+            Any object with a find_embedding(S,T,**params) method.
+            Where S and T are edgelists or NetworkX Graphs.
+
+       embedding_parameters (dict, optional, default={}):
+            Parameter dictionary to be passed to the embedding method.
+
+    """
+    def __init__(self, child_sampler, embedding_method=minorminer,
+                 **embedding_parameters):
+        if not isinstance(child_sampler, dimod.Structured):
+            raise dimod.InvalidComposition("ParamEmbeddingComposite should "
+                                           "only be applied to a "
+                                           "Structured sampler")
+        self._children = [child_sampler]
+        self._embedding_method = embedding_method
+        self._embedding_parameters = embedding_parameters
+
+        self.embedding = None
+        self.child_response = None
+
+    @property
+    def children(self):
+        """list [child_sampler]: List containing the structured sampler."""
+        return self._children
+
+    @property
+    def parameters(self):
+        """dict[str, list]: Parameters in the form of a dict.
+
+        #TODO: Ideally, the `embedding_method` object should also have a
+        `parameters` attribute.
+        """
+
+        param = self.child.parameters.copy()
+        param['force_embed'] = []
+        param['chain_strength'] = []
+        param['chain_break_method'] =  []
+        param['chain_break_fraction'] = []
+        # Ideally
+        # param['embedding_parameters'] = self._embedding_method.parameters.copy()
```

Comment:
We could also do this with [inspect](https://docs.python.org/3/library/inspect.html).

Reply:
Ah, I see the comment about cython and inspect in https://github.com/dwavesystems/minorminer/pull/82